posts/tests.py

```diff
@@ -0,0 +1,312 @@
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert file in PollFile.objects.all()
+    user.delete()
+    assert poll not in Poll.objects.all()
+    assert user not in User.objects.all()
+    assert file not in PollFile.objects.all()
+
+
+# Check if Poll can have several PollFiles simultaneously.
+@pytest.mark.django_db()
+def test_several_pollfiles(new_poll):
+    user = new_poll.author
+    poll = new_poll
+    file1 = PollFile(poll=poll, file=FILE1)
+    assert len(PollFile.objects.all()) == 0
+    user.save()
+    poll.save()
+    file1.save()
+    assert len(PollFile.objects.all()) == 1
+    poll.pollfile_set.create(file=FILE2)
+    assert len(PollFile.objects.all()) == 2
+    assert len(poll.pollfile_set.all()) == 2
+
+
+# verify that it is impossible to create a PollFile with unvalid params.
+def test_pollfile_unvalid_args():
+    with pytest.raises(ValueError):
+        PollFile(poll=USERNAME, file=FILE1)
+
+
+# -------------------------------------------- Choice tests --------------------------------------------
+
+
+# Check that the Choice values are the same as the Choice inputs.
+def test_new_choice(new_choice):
+    assert new_choice.poll.author.username == USERNAME
+    assert new_choice.poll.description == DESCRIPTION
+    assert new_choice.poll.amount_of_voters == 0
+    assert new_choice.choice_text == CHOICETEXT1
+
+
+# Check if the Choice is saved in the database and accessible via his Poll.
+@pytest.mark.django_db()
+def test_persist_choice(new_choice):
+    assert new_choice not in Choice.objects.all()
+    new_choice.poll.author.save()
+    new_choice.poll.save()
+    new_choice.save()
+    assert new_choice in Choice.objects.all()
+    assert new_choice in new_choice.poll.choice_set.all()
+
+
+# Check if Choice deletion delete only Choice from database.
+@pytest.mark.django_db()
+def test_delete_choice(new_choice):
+    user = new_choice.poll.author
+    poll = new_choice.poll
+    choice = new_choice
+    user.save()
+    poll.save()
+    choice.save()
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert choice in Choice.objects.all()
+    choice.delete()
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert choice not in Choice.objects.all()
+
+
+# Check if Choice's poll deletion delete both Choice and poll from database - but not the poll's author.
+@pytest.mark.django_db()
+def test_delete_choice_poll(new_choice):
+    user = new_choice.poll.author
+    poll = new_choice.poll
+    choice = new_choice
+    user.save()
+    poll.save()
+    choice.save()
+    assert poll in Poll.objects.all()
+    assert user in User.objects.all()
+    assert choice in Choice.objects.all()
+    poll.delete()
+    assert poll not in Poll.objects.all()
+    assert user in User.objects.all()
+    assert choice not in Choice.objects.all()
+
+
+# Check if Choice's poll's author deletion delete also Choice and database.
```

Comment:
Fix it to: "...Choice's poll author..." instead of: "...Choice's poll's author...".